service/history/engine/engineimpl/reset_queues.go

```diff
@@ -24,21 +24,22 @@ package engineimpl
 import (
 	"context"
 
+	"github.com/uber/cadence/common/persistence"
 	"github.com/uber/cadence/service/history/queue"
 )
 
 func (e *historyEngineImpl) ResetTransferQueue(
 	ctx context.Context,
 	clusterName string,
 ) error {
-	_, err := e.txProcessor.HandleAction(ctx, clusterName, queue.NewResetAction())
+	_, err := e.queueProcessors[persistence.HistoryTaskCategoryTransfer].HandleAction(ctx, clusterName, queue.NewResetAction())
```

Comment:
let's do existence check on `queueProcessors` map as it's done in other places. 
same below

Reply:
good catch, I missed this one.